### PR TITLE
Create `EnumeratorContext` 

### PR DESCRIPTION
**Pull request by kendrickt**
### Description
Creates an `EnumeratorContext`  class and add it to `Block`.

### Issue(s)
Progress towards #1083

### CODE REVIEW

**Review by natsid — APPROVED** [review]
Looks goood! Just added some suggestions around docstrings and variable naming.

File: universal-application-tool-0.0.1/app/services/applicant/RepeatedEntity.java [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
    * The {@link services.question.types.QuestionType#ENUMERATOR} question definition associated with
    * this repeated entity.
    */
-  public abstract QuestionDefinition enumeratorQuestionDefinition();
+  public abstract EnumeratorQuestionDefinition enumeratorQuestionDefinition();
 
   /** The entity name provided by the applicant. */
   public abstract String entityName();
+
+  /** The positional index with respect to its siblings. */
```
[the remark this answers is not in the document]
Comment (kendrickt) [reply universal-application-tool-0.0.1/app/services/applicant/RepeatedEntity.java:25]: added, with some more adds

File: universal-application-tool-0.0.1/app/services/applicant/Block.java [hunk 2]
```diff
@@ -111,7 +116,9 @@ public boolean isFileUpload() {
                   .map(
                       questionDefinition ->
                           new ApplicantQuestion(
-                              questionDefinition, applicantData, contextualizedPath))
+                              questionDefinition,
+                              applicantData,
+                              enumeratorContext.contextualizedPath()))
```
[the remark this answers is not in the document]
Comment (kendrickt) [reply universal-application-tool-0.0.1/app/services/applicant/Block.java:121]: Yeah. With this PR we're like 80% there, and the follow up will be actually getting this to where it needs to be used, and using it.

File: universal-application-tool-0.0.1/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java [hunk 3]
```diff
@@ -164,33 +161,37 @@ public boolean preferredLanguageSupported() {
...
 
         // Get all the repeated entities enumerated by this enumeration block.
-        QuestionDefinition enumerationQuestionDefinition =
+        EnumeratorQuestionDefinition enumerationQuestionDefinition =
             blockDefinition.getEnumerationQuestionDefinition();
-        Path contextualizedPathForEnumeration =
-            contextualizedPath.join(enumerationQuestionDefinition.getQuestionPathSegment());
-        ImmutableList<String> repeatedEntityNames =
-            applicantData.readRepeatedEntities(contextualizedPathForEnumeration);
+        Path enumPath =
```
[the remark this answers is not in the document]
Comment (kendrickt) [reply universal-application-tool-0.0.1/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java:166]: fixed

File: universal-application-tool-0.0.1/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java [hunk 4]
```diff
@@ -164,33 +161,37 @@ public boolean preferredLanguageSupported() {
       if (blockDefinition.isEnumerator()) {
 
         // Get all the repeated entities enumerated by this enumeration block.
-        QuestionDefinition enumerationQuestionDefinition =
+        EnumeratorQuestionDefinition enumerationQuestionDefinition =
```
[the remark this answers is not in the document]
Comment (kendrickt) [reply universal-application-tool-0.0.1/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java:164]: oops!!! good catch.

**Comment by kendrickt** [comment]
Closing this in favor of putting it all in `RepeatedEntity`.

**Pull request closed without merging**